patchwork/steps/CreatePR/CreatePR.py

```diff
@@ -148,20 +151,26 @@ def create_pr(
     target_branch_name: str,
     scm_client: ScmPlatformClientProtocol,
     force: bool = False,
+    issue_url: Optional[str] = None,  # Optional GitHub Issue URL to link the PR to
 ):
     prs = scm_client.find_prs(repo_slug, original_branch=base_branch_name, feature_branch=target_branch_name)
     pr = next(iter(prs), None)
     if pr is None:
+        final_body = body
+        if issue_url is not None:
+            issue_info = scm_client.get_slug_and_id_from_url(issue_url)
+            if issue_info is not None:
+                _, issue_number = issue_info
+                final_body = f"{body}\n\nResolves #{issue_number}"
+
```

Comment:
This is only applicable for github repos. Perhaps check the scm_client's type first before performing this.